official/projects/lra/exponential_moving_average.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Keras-based MegaEncoder block layer."""
+
+from typing import Optional
+import tensorflow as tf
+
+
+class MultiHeadEMA(tf.keras.layers.Layer):
+  """Exponential Moving Average Layer.
+
+    See "https://arxiv.org/abs/2209.10655" for more details.
+    """
+
+  def __init__(
+      self,
+      embed_dim,
+      ndim=2,
+      bidirectional=False,
+      truncation=None,
+  ):
```

Comment:
add **kwargs and pass to super.__init__(). There are some keras layer defaults like `name`.

Reply:
I see, edits made